src/sid/initial_conditions.py

```diff
@@ -449,29 +451,26 @@ def _sample_factorized_virus_strains_for_infections(
 
 
 def _integrate_immune_individuals(
-    states: pd.DataFrame, initial_immunity: pd.Series
+    states: pd.DataFrame,
+    initial_immunity: pd.Series,
+    number_burn_in_periods: int,
 ) -> pd.DataFrame:
     """Integrate immunity level of individuals in states.
 
-
-    TODO: The case where initial immunity is due to vaccination is  # noqa: T000
-    ignored here. This does not seem right?
-
     Args:
         states (pandas.DataFrame): The states which already include sampled infections.
         initial_immunity (pandas.Series): A series with sampled immunity levels of
             individuals.
+        number_burn_in_periods (int): The number of periods over which infections are
+            distributed and can progress. The default is one period.
 
     Returns:
         states (pandas.DataFrame): The states with initial immunity integrated.
 
     """
-    immunity = np.maximum(initial_immunity, states["immunity"])
-    states["immunity"] = immunity
-
-    locs = immunity > 0
-    states.loc[locs, "ever_infected"] = True
-    states.loc[locs, "cd_ever_infected"] = 0
-    states.loc[locs, "cd_immune_false"] = states.loc[locs, "cd_immune_false_draws"]
+    extra_immune = initial_immunity > states["immunity"]
+    states.loc[extra_immune, "ever_infected"] = True
+    states.loc[extra_immune, "cd_ever_infected"] = number_burn_in_periods + 1
```

Comment:
This needs to be a negative number. `number_burn_in_periods + 1` have **passed** since the `ever_infected` countdown should have started.

Reply:
Done.